### ISSUE TITLE
(feature request) Case check for Comp.propTypes

### ISSUE THREAD

**Issue by SeanSilke** [yannickcr/eslint-plugin-react#1420]
If misplaced lowercase `p` for `P` in `propTypes` it will lead to complete absence of props type checking

It will be good to add check for  `Comp.propTypes` vs ` Comp.PropTypes`

How strange it case can seem people encountered it.
[example on stackoverflow](https://stackoverflow.com/questions/45398709/react-proptypes-in-stateless-functional-component)

**Comment by jseminck** [comment]
https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-typos.md handles this. However the rule has been a bit unstable 😞 the `7.4.0-rc0` release should be good though. That reminds me, that I should check it.

**Comment by SeanSilke** [comment]
@jseminck Thanks will definitely use it.

**Comment by SeanSilke** [comment]
@jseminck Tried  `7.4.0-rc0`. 
Unfortunately it does not work for this case.
No warnings on misspelled `.PropTypes`

**Comment by jseminck** [comment]
Can you give an example? This is exactly what the rule is supposed to do - so perhaps there's a case that isn't caught properly.

![proptypes](https://user-images.githubusercontent.com/9586897/30237525-a5da9c96-953c-11e7-8fc7-e4719f1a96bd.png)

**Comment by SeanSilke** [comment]
@jseminck Oh, my bad.
I have typo in property of stateless functional component.
<img width="408" alt="screenshot at sep 09 14-03-00" src="https://user-images.githubusercontent.com/1077074/30239545-9dac3e78-9567-11e7-84df-fd790d7075f6.png">

This rule is only for static property of a class.

Will it be good to give this rule less broad name?
Is it possible to extend this rule and add check for typos in properties of stateless functional component?

**Comment by jseminck** [comment]
It should work as well. At least we have test cases for it... I will try to
figure out why your example isn't working.

**Comment by jseminck** [comment]
I tried your example in our test suite and it correctly identifies it. (note: I added it as a valid test case so it would throw an error)

```
1) no-typos valid
      export const HighlightTextInner = (
        {muiTheme: {palette: {accent3Color}}, ...rest}
      ) =>
        <span {...rest} style={{color: accent3Color}} />;

      HighlightTextInner.PropTypes = {
        children: PropTypes.string.isRequired
      }
    :

      AssertionError: Should have no errors but had 1: [ { ruleId: 'no-typos',
    severity: 1,
    message: 'Typo in static class property declaration',
    line: 7,
    column: 7,
    nodeType: 'MemberExpression',
    source: '      HighlightTextInner.PropTypes = {',
    endLine: 7,
    endColumn: 35 } ]
      + expected - actual

      -1
      +0
```

**Comment by SeanSilke** [comment]
@jseminck 
Yes it Works!
I made errors in declaration on the rule.
Great thanks!

There is possibility to make better documentation.
### Add functional components to  warning code examples.
```js
const Welcome = ({name}) => <h1> Hello, {name} </h1>
Welcome.PropTypes = {}

const Welcome = ({name}) => <h1> Hello, {name} </h1>
Welcome.proptypes = {}
```
### Add mention of function in rule overview.
Ensure no casing typos were made declaring static class properties
-> Ensure no casing typos were made declaring static class or **function** properties

 
It makes sure that the following class properties have no casing typos
-> It makes sure that the following class of **function** properties have no casing typos

Once again thanks for your help!

**Comment by ljharb** [comment]
Sounds like this is resolved; PRs to improve docs are always appreciated.
